ff-netF48-server-sdk.csproj

```diff
@@ -38,7 +38,8 @@
         <PackageReference Include="murmurhash" Version="1.0.3" />
         <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
         <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
-        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" />
+        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'net461' Or '$(TargetFramework)' == 'net48' Or '$(TargetFramework)' == 'net5.0' Or '$(TargetFramework)' == 'net6.0' Or '$(TargetFramework)' == 'net7.0'" />
```

Comment:
I think this should be simply != net8.0 - the conditions on the System.Net.Http ref are explicit due to the fact that there is a strict whitelist pof places where it's necessary in order for things to compile

This would also be as likely to lead to the right thing happening when a net10.0 TFM variant gets added

Reply:
You may already be aware of this, however I decided to do some testing with the following PackageReference configurations and discovered some interesting behavior. I was concerned that if I just specified != 'net8.0' that it may cause a conflict when generating the nuget package. This is partially true but depends on a couple of factors.

In this example I used net7.0 instead of net9.0 as .NET 9 is in RC status and I wanted to avoid playing around with beta nuget packages.

If I have the PackageReference defined as follows, the generated nuget package grabs System.IdentityModel.Tokens.Jwt version 6.x and bundles that in with the .NET 7 library dependency collection packaged in the nuget file.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
```

![2024-09-19_08-58-13](https://github.com/user-attachments/assets/52c999b8-0cbc-4436-9a3f-021b1209631a)

This next example correctly bundles System.IdentityModel.Tokens.Jwt with the 7.x version and places it in the correct dependencies path in the package.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net7.0' And '$(TargetFramework)' != 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
```

![2024-09-19_08-56-13](https://github.com/user-attachments/assets/35ca1089-81d7-45f2-86ba-6a0ee0c4f4fb)

Lastly, It appears that order is important when using the PackageReference conditions. This also appears to correctly bundle the System.IdentityModel.Tokens.Jwt version 7.x with the nuget package.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
```

By placing the newest versions first, with the excplicit version condition it grabs the one in order based on the version of .NET being built and then bundles that library correctly in the nuget package file.

I have updated the csproj file to follow this last example, making sure that the net8.0 version is first in the list. I also wanted to caution a careful review of your library package prior to publishing to make sure the correctly expected System.IdentityModel.Tokens.Jwt version is included in later versions such as when you include net9.0 later. I will push these updates shortly.

I would suggest when later adding the 9.x version of the library, to place it above the net8.0 so as to correctly include the expected library dependencies in order.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="9.0.0" Condition="'$(TargetFramework)' == 'net9.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
```

Reply:
You may already be aware of this, however I decided to do some testing with the following PackageReference configurations and discovered some interesting behavior. I was concerned that if I just specified != 'net8.0' that it may cause a conflict when generating the nuget package. This is partially true but depends on a couple of factors.

In this example I used net7.0 instead of net9.0 as .NET 9 is in RC status and I wanted to avoid playing around with beta nuget packages.

If I have the PackageReference defined as follows, the generated nuget package grabs System.IdentityModel.Tokens.Jwt version 6.x and bundles that in with the .NET 7 library dependency collection packaged in the nuget file.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
```

![2024-09-19_08-58-13](https://github.com/user-attachments/assets/e5f7a983-bedc-40a9-89ad-1ca4f5b952d6)

This next example correctly bundles System.IdentityModel.Tokens.Jwt with the 7.x version and places it in the correct dependencies path in the package.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net7.0' And '$(TargetFramework)' != 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
```

![2024-09-19_08-56-13](https://github.com/user-attachments/assets/fe726817-b42c-4493-a305-5250ca1bd9ce)

Lastly, It appears that order is important when using the PackageReference conditions. This also appears to correctly bundle the System.IdentityModel.Tokens.Jwt version 7.x with the nuget package.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.7.1" Condition="'$(TargetFramework)' == 'net7.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
```

By placing the newest versions first, with the excplicit version condition it grabs the one in order based on the version of .NET being built and then bundles that library correctly in the nuget package file.

I have updated the csproj file to follow this last example, makeing sure that the net8.0 version is first in the list. I also wanted to caution a careful review of your library package prior to publishing to make sure the correctly expected System.IdentityModel.Tokens.Jwt version is included in later versions such as when you include net9.0 later.

I would suggest when later adding the 9.x version of the library, to place it above the net8.0 so as to correctly include the expected library dependencies in order.

```
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="9.0.0" Condition="'$(TargetFramework)' == 'net9.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.0.2" Condition="'$(TargetFramework)' == 'net8.0'" />
        <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.34.0" Condition="'$(TargetFramework)' != 'net8.0'" />
```

Reply:
all sounds good - would put your masterplan in XML comments over the PackageRefrernces block

On reflection you were on the right track at the outset though, i.e. probably better to default to no reference unless its intentional
so i.e. have a "if netstandard OR .... net6.0" then <one it had>
if == net70 the. net70 ver
if == net80 the. net80 ver
then when soeone adds net9.0/net10.0 to TFM there will be a missing reference and they will be made to think

Apologies for the misdirection!

Reply:
You bet, I'll get that updated.

Reply:
I just added a comment above that should hopefully communicate the dependency order needed when updating nuget packages.

Reply:
Or would you like me to update that last one to use the same Conditional technique as System.Net.Http for the Tokens.Jwt library?

Reply:
I think my original stance/point was relevant there - it was only ever needed for in some specific cases. My main objective was to remove a noisy/egregious dependency. The fact that the conditional defaulys to omitting, and that the compiler will tell us if that is wrong makes it realobly safe and obvious IMO, but up to you ;)

Reply:
I'm happy to update it in the way you feel is best and easiest to maintain.

Reply:
for avoidance of doubt I'm a random overopinionanted drive by contributor to the repo - Erdi will/can make any actual calls; please don't feel compelled to do/not do anything on my say so (but what you're doing is all good and I'm glad you're doing this stuff as I'll benefit from it in some future round of package updates my side 🙏)